client/src/pages/components/Pagination.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+
+const Pagination = ({ postsPerPage, totalPosts, paginate }) => {
+	const pageNumbers = [];
+
+	for (let i = 1; i <= Math.ceil(totalPosts / postsPerPage); i++) {
```

Comment:
https://stackoverflow.com/questions/3746725/how-to-create-an-array-containing-1-n

Reply:
?

Not sure whether its related or any important but I have figured that postsPerPage has to be a positive integer greater than 0 otherwise it breaks the page, so I added a comment in the component where it is initiated about it and also a pretty rubbish validation rule in the Pagination component, I am sure there are better ways to go about it;